Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -318,7 +338,9 @@ private async Task<UriBuilder> UploadBlobChunkedAsync(string repository, string
             // Fail the upload if the response code is not Accepted (202) or if uploading to Amazon ECR which returns back Created (201).
             if (!(patchResponse.StatusCode == HttpStatusCode.Accepted || (IsAmazonECRRegistry && patchResponse.StatusCode == HttpStatusCode.Created)))
             {
-                string errorMessage = Resource.FormatString(nameof(Strings.BlobUploadFailed), patchUri, patchResponse.StatusCode, await patchResponse.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false));
+                var headers = patchResponse.Headers.ToString();
+                var detail = await patchResponse.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
+                string errorMessage = Resource.FormatString(nameof(Strings.BlobUploadFailed), $"Chunked PATCH {patchUri}", patchResponse.StatusCode, headers + Environment.NewLine + detail);
```

Comment:
I started logging headers and the content, since headers often contain some kind of tracking IDs for error reporting purposes. I also added the kind of interaction with the API, since that wasn't clear entirely from Uri.